File: radialbuttonmenu/src/main/java/huberd/laurent/radialbuttonmenu/Animation/BasicAnimationManager.java
package huberd.laurent.radialbuttonmenu.Animation;

import android.support.design.widget.FloatingActionButton;
import android.view.animation.OvershootInterpolator;

import huberd.laurent.radialbuttonmenu.Translation.AngleInterval;
import huberd.laurent.radialbuttonmenu.Translation.Translation;

import static android.R.attr.radius;

public class BasicAnimationManager implements AnimationManager {

    final protected int mainButtonRotationAngle = 225;
    protected int duration = 300;

    public BasicAnimationManager(int duration) {
        this.duration = duration;
    }

    public BasicAnimationManager() {

    }

    @Override
    public Animation getShowMainButtonAnimation(FloatingActionButton floatingActionButton, Translation translation) {
        Animation animation = new Animation(mainButtonRotationAngle, duration);
        animation.setRotationInterpolator(new OvershootInterpolator(2.0f));
        return animation;
    }

    @Override
    public Animation getHideMainButtonAnimation(FloatingActionButton floatingActionButton, Translation translation) {
        return getShowMainButtonAnimation(floatingActionButton, translation).reverse();
    }

    @Override
    public Animation getShowSecondaryButtonAnimation(int index, FloatingActionButton floatingActionButton, Translation translation) {
        Animation animation = new Animation(0, 300);
        animation.setTranslationInterpolator(new OvershootInterpolator(1.0f));
        return animation;
    }

    @Override
    public Animation getHideSecondaryButtonAnimation(int index, FloatingActionButton floatingActionButton, Translation translation) {
        Animation animation = new Animation(0, 300);
        animation.setTranslationInterpolator(new OvershootInterpolator(1.5f));
        return animation;
    }
}

File: radialbuttonmenu/src/main/java/huberd/laurent/radialbuttonmenu/Animation/BasicAnimationManager.kt
package huberd.laurent.radialbuttonmenu.Animation

import android.support.design.widget.FloatingActionButton
import android.view.animation.OvershootInterpolator
import huberd.laurent.radialbuttonmenu.Translation.Translation

data class BasicAnimationManager(
        val mainButtonRotationAngle: Int = 225,
        var duration: Int = 300
) : AnimationManager {

    override fun getShowMainButtonAnimation(floatingActionButton: FloatingActionButton, translation: Translation): Animation
        = Animation(rotation = mainButtonRotationAngle, duration = duration, rotationInterpolator = OvershootInterpolator(2.0f))

    override fun getHideMainButtonAnimation(floatingActionButton: FloatingActionButton, translation: Translation): Animation
        = getShowMainButtonAnimation(floatingActionButton, translation).reverse()

    override fun getShowSecondaryButtonAnimation(index: Int, floatingActionButton: FloatingActionButton, translation: Translation): Animation
        = Animation(rotation = 0, duration = 300, translationInterpolator = OvershootInterpolator(1.0f))

    override fun getHideSecondaryButtonAnimation(index: Int, floatingActionButton: FloatingActionButton, translation: Translation): Animation
        = Animation(rotation = 0, duration = 300, translationInterpolator = OvershootInterpolator(1.5f))
}
